projects/plugins/boost/app/assets/src/js/features/speed-score/lib/hooks.ts

```diff
@@ -93,32 +95,24 @@ export const useDebouncedRefreshScore = (
 	{ moduleStates, criticalCssCreated, criticalCssIsGenerating }: RefreshDependencies,
 	loadScore: RefreshFunction
 ) => {
-	const [ scoreConfigString, setScoreConfigString ] = useState(
-		JSON.stringify( [ moduleStates, criticalCssCreated ] )
-	);
+	const lastScoreConfigString = useRef( JSON.stringify( [ moduleStates, criticalCssCreated ] ) );
 
 	// Debounced function: Refresh the speed score if the config has changed.
-	const debouncedRefreshScore = useMemo(
-		() =>
-			debounce( async ( newConfig, oldConfig ) => {
-				if ( oldConfig !== newConfig ) {
-					setScoreConfigString( newConfig );
-					await loadScore();
-				}
-			}, 2000 ),
-		[ loadScore, setScoreConfigString ]
+	// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I think we should establish that whenever we silence this eslint - there should be a paragraph long explanation exactly why that is okay.